public/resources/core/js/services/product-offering.service.js

```diff
@@ -97,53 +118,117 @@
             return this;
         };
         Price.prototype.toJSON = function toJSON() {
-            return {
-                currencyCode: this.currencyCode,
+            var data = {
                 dutyFreeAmount: this.taxIncludedAmount / ((100 + this.taxRate) / 100),
                 percentage: this.percentage,
                 taxIncludedAmount: this.taxIncludedAmount,
                 taxRate: this.taxRate
             };
+
+            if ('currencyCode' in this) {
+                data.currencyCode = this.currencyCode;
+            }
```

Comment:
Have you tested that this works when the price is part of an alteration, because the alteration price does not include a currencyCode in the RI